src/systemctl/systemctl.c

```diff
@@ -3536,7 +3535,13 @@ static int load_kexec_kernel(void) {
                 _exit(EXIT_FAILURE);
         }
 
-        return wait_for_terminate_and_check("kexec", pid, WAIT_LOG);
+        r = wait_for_terminate_and_check("kexec", pid, WAIT_LOG);
+        if (r < 0)
+                return r;
+        if (r > 0)
+                /* Command failed */
+                return -EINVAL;
```

Comment:
not that it matters, but we usually use EPROTO for propagating failures due to non-zero child process return values...